rts/motoko-rts/src/custom.rs

```diff
@@ -0,0 +1,326 @@
+// Custom RTS function utilities
+
+use alloc::vec::Vec;
```

Comment:
This would potentially involve dynamic allocation using the Rust-interop `EmphermalAllocator` in `allocator.rs` allocating `Blob` behind the scenes. Did you check how many implicit allocations happen? I wonder whether we could possibly avoid the `Vec` and use custom convenience wrapper around `Blob` and `Array` for implementing FFI functions.